pkg/opensearch/ppl_response_parser.go

```diff
@@ -53,7 +55,69 @@ func (rp *pplResponseParser) parseTimeSeries() (*backend.DataResponse, error) {
 		Frames: data.Frames{},
 	}
 
-	t, err := getResponseMeta(rp.Response.Schema)
+	switch rp.Query.Format {
+	case logsType:
+		return rp.parseLogs(queryRes, configuredFields)
+	}
+	return rp.parseTimeSeries(queryRes)
+}
+
+func (rp *pplResponseParser) parseLogs(queryRes *backend.DataResponse, configuredFields es.ConfiguredFields) (*backend.DataResponse, error) {
+	propNames := make(map[string]bool)
+	docs := make([]map[string]interface{}, len(rp.Response.Datarows))
+
+	for rowIdx, row := range rp.Response.Datarows {
+		doc := map[string]interface{}{}
+
+		for fieldIdx, field := range rp.Response.Schema {
+			value := row[fieldIdx]
+			fieldType := field.Type
+
+			if fieldType == "timestamp" || fieldType == "datetime" || fieldType == "date" {
+				timestampFormat := pplTSFormat
+				if fieldType == "date" {
+					timestampFormat = pplDateFormat
+				}
+				ts, err := rp.parseTimestamp(row[fieldIdx], timestampFormat)
+				if err != nil {
+					return nil, err
+				}
+				value = *utils.NullFloatToNullableTime(ts)
+			}
+
+			doc[field.Name] = value
+		}
+
+		if configuredFields.LogLevelField != "" {
+			doc["level"] = doc[configuredFields.LogLevelField]
+		}
+
+		doc = flatten(doc, maxFlattenDepth)
+		for key := range doc {
+			if _, ok := propNames[key]; !ok {
+				propNames[key] = true
+			}
+		}
+
+		docs[rowIdx] = doc
+	}
+
+	sortedPropNames := sortPropNames(propNames, []string{configuredFields.TimeField, configuredFields.LogMessageField})
+	fields := processDocsToDataFrameFields(docs, sortedPropNames)
+
+	frame := data.NewFrame("", fields...)
+	if frame.Meta == nil {
+		frame.Meta = &data.FrameMeta{}
+	}
+	frame.Meta.PreferredVisualization = data.VisTypeLogs
+
+	queryRes.Frames = append(queryRes.Frames, frame)
+
+	return queryRes, nil
+}
```

Comment:
I wonder to what extent we should try to mutualize some of this code with the Lucene code, what do you think? (maybe in a following PR)

I'd like to avoid a situation where we have to make changes on "both sides." At the same time I don't want to introduce code that is too rigid.

Reply:
this particular part of the code that processes logs is relatively short so i don't think there's too much of a problem with having to make changes to both sides.

we can look at sharing some of the code with the lucene logs side though. i think after finishing the rest of the query migrations would be a good time to do so, maybe when we are actually working on any changes to the logs processing.